system/Commands/Translation/LocalizationSync.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Commands\Translation;
+
+use CodeIgniter\CLI\BaseCommand;
+use CodeIgniter\CLI\CLI;
+use CodeIgniter\Exceptions\LogicException;
+use Config\App;
+use Locale;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use SplFileInfo;
+
+/**
+ * @see \CodeIgniter\Commands\Translation\LocalizationSyncTest
+ */
+class LocalizationSync extends BaseCommand
+{
+    protected $group       = 'Translation';
+    protected $name        = 'lang:sync';
+    protected $description = 'Synchronize translation files from one language to another.';
+    protected $usage       = 'lang:sync [options]';
+    protected $arguments   = [];
+    protected $options     = [
+        '--locale' => 'The original locale (en, ru, etc.).',
+        '--target' => 'Target locale (en, ru, etc.).',
+    ];
+    private string $languagePath;
+
+    public function run(array $params)
+    {
+        $optionTargetLocale = '';
+        $optionLocale       = $params['locale'] ?? Locale::getDefault();
+        $this->languagePath = APPPATH . 'Language';
+
+        if (isset($params['target']) && $params['target'] !== '') {
+            $optionTargetLocale = $params['target'];
+        }
+
+        if (! in_array($optionLocale, config(App::class)->supportedLocales, true)) {
+            CLI::error(
+                'Error: "' . $optionLocale . '" is not supported. Supported locales: '
+                . implode(', ', config(App::class)->supportedLocales)
+            );
+
+            return EXIT_USER_INPUT;
+        }
+
+        if ($optionTargetLocale === '') {
+            CLI::error(
+                'Error: "--target" is not configured. Supported locales: '
+                . implode(', ', config(App::class)->supportedLocales)
+            );
+
+            return EXIT_USER_INPUT;
+        }
+
+        if (! in_array($optionTargetLocale, config(App::class)->supportedLocales, true)) {
+            CLI::error(
+                'Error: "' . $optionTargetLocale . '" is not supported. Supported locales: '
+                . implode(', ', config(App::class)->supportedLocales)
+            );
+
+            return EXIT_USER_INPUT;
+        }
+
+        if ($optionTargetLocale === $optionLocale) {
+            CLI::error(
+                'Error: You cannot have the same values "--target" and "--locale".'
+            );
+
+            return EXIT_USER_INPUT;
+        }
+
+        if (ENVIRONMENT === 'testing') {
+            $this->languagePath = SUPPORTPATH . 'Language';
+        }
+
+        $this->process($optionLocale, $optionTargetLocale);
```

Comment:
If this method encounters a failure (e.g., via `CLI::error()`), we should not exit successfully later. We should instead use `EXIT_ERROR`. I think this process method should not return void but return an int code, check it here, if fail return early with exit failure.